src/server/graphics/platform_probe.cpp

```diff
@@ -168,22 +179,32 @@ auto modules_for_device(
                         best_modules_so_far.emplace_back(std::move(device), module);
                     }
                 }
-                else if (device.support_level > mg::probe::unsupported)
+                else
                 {
-                    // Devices with null associated udev device are not combined with any others
-                    best_modules_so_far.emplace_back(std::move(device), module);
+                    /* Platforms with no associated `mir::udev::Device` are *nested* platforms
+                     * We want at most one of these
```

Comment:
We have `probe::hosted`: is using both "hosted" and "nested" confusing?

Reply:
Hm. There's no longer really any need for `probe::hosted` *at all*; we they're on a nested/hardware is now a separate axis to unsupported/best.

Should we *remove* `probe::hosted`?

Reply:
Or just stick to `probe::hosted`, rather than `::nested`?

Reply:
Well, _if_ the intention of this PR is to change the terminology from "hosted" to "nested" it makes sense to remove (or rename) `probe::hosted`.

But I've memories of long ago replacing "nested" with "hosted" and wonder if we want revisit that again?